### ISSUE TITLE
Add support for `styled('element')` syntax

### ISSUE THREAD

**Issue by EvgenyOrekhov** [ota-meshi/eslint-plugin-css#84]
**Description**

MUI uses `styled('element')` instead of `styled.element`, but otherwise the functionality is almost the same.
It would be great if `styled('element')` syntax was supported.

```ts
import { styled } from '@mui/system';

const MyComponent = styled('div')({
  color: 'darkslategray',
  backgroundColor: 'aliceblue',
  padding: 8,
  borderRadius: 4,
});
```

**Comment by EvgenyOrekhov** [comment]
I can see the above example works if I use `import styled from 'styled-components'`.
So it should be just a matter of detecting `import { styled } from '@mui/system'`, right?

**Comment by ota-meshi** [comment]
Yeah. We probably need to extend the default behavior of `target.defineFunctions`.

https://ota-meshi.github.io/eslint-plugin-css/settings/#target-definefunctions

**Comment by joshsylvester** [comment]
one comment on the above - and related to styled components that i'd like to highlight

```
  padding: 8,
  borderRadius: 4,
```

the shorthand suggested by the OP (not `"8px"`, but `8`) syntax would be nice to support across the board (not just in styled components, but inline styling as well) - adding a rule that checks for properties that don't need "px" values and be able to warn or enforce that syntax. 

Other systems like MUI that use object based css properties would be great to support as well. ie: [`sx`](https://mui.com/system/getting-started/the-sx-prop/) & [`makeStyles`](https://mui.com/system/styles/api/#makestyles-styles-options-hook)

**Comment by litera** [comment]
Not to forget other possible style definitions in MUI.

### Breakpoints support
```
sx={{ margin: { xs: 1, sm: 1.5, md: 2 } }}
```
### CSS properties shorthand names
```
sx={{ m: { xs: 1, sm: 1.5, md: 2 } }}
```
### Nested selectors
```
sx={{ color: '#f00', ':hover': { color: '#0f0' } }}
```
### Named colors
```
sx={{ color: 'primary.main' }}
```
### Function style definitions
```
sx={({ theme}) => ({ margin: theme.spacing(1, 2) })}
// or
sx={{ margin: ({ theme}) => theme.spacing(1, 2) }}
```
### Array definitions
```
sx={[ { m: 1 }, isError && { color: '#f00' }, ({ theme }) => ({ p: theme.spacing(1) }) ]}
```
### Component theming
```
theme = {
    MuiButton: {
        styleOverrides: {
            root: {
                // all style definitions
            },
            secondary: ({theme}) => ({
                // all style definitions
            })
        }
    }
}
```

etc.

I'm wondering whether this particular ESLint plugin is capable of all of these and if there is maybe some other existing plugin that gives us all of these out of the box?

It would be even better if it supported browserslist, so that CSS would be checked against CSS feature support.

I'm using `eslint-plugin-compat` for Typescript feature support and I'd like to add CSS-in-JS support for MUI themed components, MUI styled components and sx prop compatibility checking.
